### PR TITLE
BUGFIX: Do not resolve inputDateTime as relation

### LINKED ISSUE #147: Tca fields with renderType set are treated as relations
Fields like 'starttime' of table 'tt_content' have a renderType="inputDateTime".

The `RelationResolver` treats this kind of fields like a relation and returns an empty array for this kind of fields instead of indexing the timestamp in this case.
https://github.com/Codappix/search_core/blob/7cc1616d3a2d6d5734fc6deb3bfeaf1dad84168e/Classes/Domain/Index/TcaIndexer/RelationResolver.php#L85

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Add test which checks this behaviour.
Exclude inputDateTime from relation resolving.

Resolves #147

### CODE REVIEW

**Comment by martinhummer** [comment]
I think the TCA Config is not correct.
I made a new version of the test, which uses the TCA config for tt_content field 'starttime' in 8.7.12 LTS:
https://github.com/martinhummer/search_core/blob/develop/Tests/Unit/Domain/Index/TcaIndexer/RelationResolverTest.php

This test will fail, because `BackendUtility::getProcessedValueExtra` returns a String like '06-04-18 10:36' and not a timestamp.

Do you think it would be possible to exclude inputDateTime fields completely from this processing and return the (int) timestamp as it is in the database.

**Comment by DanielSiepmann** [comment]
I can't see any difference of the TCA Configuration. Without the change to skip the renderType I'll receive an empty value for `starttime` in the test. With the change I'll receive the original value. Perhaps we can chat on that to make sure it works as expected?

File: Tests/Unit/Domain/Index/TcaIndexer/RelationResolverTest.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        $tableServiceMock->expects($this->any())
+            ->method('getColumnConfig')
+            ->willReturn($GLOBALS['TCA']['tt_content']['columns']['starttime']['config']);
+
+        $this->subject->resolveRelationsForRecord($tableServiceMock, $record);
+
+        $this->assertSame(
+            $originalRecord,
+            $record,
+            'starttime'
```
Comment (justusmoroni) [inline Tests/Unit/Domain/Index/TcaIndexer/RelationResolverTest.php]: Shouldn't there be a more understandable message?

Comment (justusmoroni) [reply Tests/Unit/Domain/Index/TcaIndexer/RelationResolverTest.php]: Like here?
https://github.com/Codappix/search_core/pull/152/files#diff-996a498714afe5c8a5c1802452779dcbR89

**Review by justusmoroni — CHANGES_REQUESTED** [review]

**Review by justusmoroni — APPROVED** [review]

**Comment by DanielSiepmann** [comment]
We will keep the test, but will not process this issue itself anymore. Instead, see the issue, we will resolve this by working on #149 .

**Pull request closed without merging**